### ISSUE TITLE
Feature: New contact request user flow

### ISSUE THREAD

**Issue by yqrashawn** [status-im/status-react#18965]
Blocked by:
- #18962
- #18964

User story:

10. **Initiating a Contact Request**
    - As a user, I want to initiate a contact request to a stranger, so that I can express my interest to connect with them for future interactions within the app.

11. **Adding a Personalized Message to Contact Request**
    - As a user, I want to add a personalized message with my contact request, so that the recipient has context on who I am and why I want to connect.

12. **Reviewing Contact Request Before Sending** 
    - As a user, I want to review my contact request and personal message before sending, so that I can make sure it contains the right information and is free of errors.

13. **Sending the Contact Request**
    - As a user, I want to send the contact request after typing my message, so that I can formally request to connect with the other user.

Scope:
- UI for the full user flow
- No toast
- Create a separate issue if the auto-expand input is not already implemented

Figma: https://www.figma.com/file/jGs1P4RU4R1NRrY0xXmHC1/Profile-interactions?type=design&node-id=0-38997&mode=design&t=L27O8eaCfS59gCGl-4

![Image](https://github.com/status-im/status-mobile/assets/15090582/6f175dea-2bca-4f43-a307-0f40b1044c48)
